crates/scmi/src/devices/common.rs

```diff
@@ -0,0 +1,552 @@
+    /// The device specification must be used in [available_devices] to
+    /// actually add the device.
+    fn new(
+        constructor: DeviceConstructor,
+        short_help: &str,
+        long_help: &str,
+        parameters_help: Vec<&str>,
+    ) -> Self {
+        Self {
+            constructor,
+            short_help: short_help.to_owned(),
+            long_help: long_help.to_owned(),
+            parameters_help: parameters_help.into_iter().map(ToOwned::to_owned).collect(),
+        }
+    }
+}
+
+/// Mapping of device identifiers (names) to device specifications.
+///
+/// The string keys correspond to device identifiers specified on the command
+/// line.
+type NameDeviceMapping = HashMap<&'static str, DeviceSpecification>;
+
+/// Creates device mapping and adds all the supported devices to it.
+///
+/// If you want to introduce a new kind of host device bindings, insert a
+/// device identifier + [DeviceSpecification] to [NameDeviceMapping] here.
+pub fn available_devices() -> NameDeviceMapping {
+    let mut devices: NameDeviceMapping = HashMap::new();
+    devices.insert(
+        "fake",
+        DeviceSpecification::new(
+            fake::FakeSensor::new_device,
+            "fake accelerometer",
+            "A simple 3-axes sensor providing fake pre-defined values.",
+            vec!["name: an optional name of the sensor, max. 15 characters"],
+        ),
+    );
+    devices.insert(
+        "iio",
+        DeviceSpecification::new(
+            iio::IIOSensor::new_device,
+            "industrial I/O sensor",
+            "",
+            vec![
+                "path: path to the device directory (e.g. /sys/bus/iio/devices/iio:device0)",
+                "channel: prefix of the device type (e.g. in_accel)",
+                "name: an optional name of the sensor, max. 15 characters",
+            ],
+        ),
+    );
+    devices
+}
+
+fn devices_help() -> String {
+    let mut help = String::new();
+    writeln!(help, "Available devices:").unwrap();
+    for (name, specification) in available_devices().iter() {
+        let short_help = &specification.short_help;
+        let long_help = &specification.long_help;
+        let parameters_help = &specification.parameters_help;
+        writeln!(help, "\n- {name}: {short_help}").unwrap();
+        for line in long_help.lines() {
+            writeln!(help, "  {line}").unwrap();
+        }
+        if !parameters_help.is_empty() {
+            writeln!(help, "  Parameters:").unwrap();
+            for parameter in parameters_help {
+                writeln!(help, "  - {parameter}").unwrap();
+            }
+        }
+    }
+    writeln!(help, "\nDevice specification example:").unwrap();
+    writeln!(
+        help,
+        "--device iio,path=/sys/bus/iio/devices/iio:device0,channel=in_accel"
+    )
+    .unwrap();
+    help
+}
+
+pub fn print_devices_help() {
+    let help = devices_help();
+    println!("{}", help);
+}
+
+pub fn check_device_properties(
```

Comment:
This should probably be moved to impl DeviceProperties instead.

Reply:
Done.